pyparsing/core.py

```diff
@@ -2331,11 +2319,9 @@ def __new__(cls, match_string: str = "", *, matchString: str = ""):
     def __init__(self, match_string: str = "", *, matchString: str = ""):
         super().__init__()
         match_string = matchString or match_string
-        if not match_string:
-            raise ValueError("Empty string should not reach Literal.__init__()")
         self.match = match_string
         self.matchLen = len(match_string)
-        self.firstMatchChar = match_string[0]
+        self.firstMatchChar = match_string[0] if match_string else ''
```

Comment:
Could also write as `self.firstMatchChar = match_string[:1]`

Reply:
Also, run `black`, which will change all string literals to using "" instead of ''